code/notificationbuilder/src/main/java/com/adobe/marketing/mobile/notificationbuilder/internal/builders/ProductCatalogNotificationBuilder.kt

```diff
@@ -0,0 +1,335 @@
+/*
+  Copyright 2024 Adobe. All rights reserved.
+  This file is licensed to you under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License. You may obtain a copy
+  of the License at http://www.apache.org/licenses/LICENSE-2.0
+  Unless required by applicable law or agreed to in writing, software distributed under
+  the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+  OF ANY KIND, either express or implied. See the License for the specific language
+  governing permissions and limitations under the License.
+*/
+
+package com.adobe.marketing.mobile.notificationbuilder.internal.builders
+
+import android.app.Activity
+import android.app.NotificationManager
+import android.app.PendingIntent
+import android.content.BroadcastReceiver
+import android.content.Context
+import android.widget.RemoteViews
+import androidx.core.app.NotificationCompat
+import com.adobe.marketing.mobile.notificationbuilder.NotificationConstructionFailedException
+import com.adobe.marketing.mobile.notificationbuilder.PushTemplateIntentConstants
+import com.adobe.marketing.mobile.notificationbuilder.R
+import com.adobe.marketing.mobile.notificationbuilder.internal.PendingIntentUtils
+import com.adobe.marketing.mobile.notificationbuilder.internal.PushTemplateConstants
+import com.adobe.marketing.mobile.notificationbuilder.internal.PushTemplateConstants.LOG_TAG
+import com.adobe.marketing.mobile.notificationbuilder.internal.PushTemplateImageUtils
+import com.adobe.marketing.mobile.notificationbuilder.internal.extensions.createNotificationChannelIfRequired
+import com.adobe.marketing.mobile.notificationbuilder.internal.extensions.setElementColor
+import com.adobe.marketing.mobile.notificationbuilder.internal.templates.ProductCatalogPushTemplate
+import com.adobe.marketing.mobile.services.Log
+
+/**
+ * Object responsible for constructing a [NotificationCompat.Builder] object containing a product catalog template notification.
+ */
+internal object ProductCatalogNotificationBuilder {
+    private const val SELF_TAG = "ProductCatalogNotificationBuilder"
+    private val intentActionList = listOf(
```

Comment:
It might be better to use the same intent action `PushTemplateIntentConstants.IntentActions.CATALOG_THUMBNAIL_CLICKED` and a different requestCode to make it easier to handle on caller extension side. Also works better if we decide to support more than 3 thumbail icons in the future

Reply:
sure i can make this change